### ISSUE TITLE
Adding a rule to check the possibility of the variable being null

### ISSUE THREAD

**Issue by EngRajabi** [meziantou/Meziantou.Analyzer#534]
Adding a rule to check the possibility of the variable being null

~~~csharp
    public void LastIndex(CancellationToken cancellationToken)
    {
        var g = GetNullable();
        Console.Write(g.Value); //possible null
    }

    private static long? GetNullable()
    {
        return null;
    }
~~~
A warning can be issued by checking Nullable<>

**Comment by meziantou** [comment]
What would be the difference with [Nullable Reference Types](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references?WT.mc_id=DT-MVP-5003978)?

**Comment by EngRajabi** [comment]
I did not catch. What I mean is to create a rule that warns about the possibility of nulling the variables.

**Comment by EngRajabi** [comment]
I tested. With the example of the code I gave above. No warning has been issued.
Of course, Nullable is disabled in my project. But the above code has the possibility of null error.

**Comment by meziantou** [comment]
> Of course, Nullable is disabled in my project

You must enable nullables using `<Nullable>enable</Nullable>` or `#nullable enable` to get the warnings.

But this doesn't answer my initial question. Why do you want me to implement a new rule? How is your request different from what the compiler already provides with `<Nullable>enable</Nullable>`?

**Comment by EngRajabi** [comment]
I don't want to use the Nullable feature both for personal reasons and because the code is old and needs a lot of changes.
There is a new rule request for projects that have Nullable disabled, but still have the possibility of a null error. be corrected
In the code above, it is correct that nullable is inactive, but there is a possibility of a null error that can be notified to the developer with a new rule.

**Comment by meziantou** [comment]
So, you want something similar to Nullable Reference Types (NRT), but without using NRT. 

Meziantou.Analyzer won't implement such rules. There is no need to reimplement something that already exists. If you don't want to use NRT, that your choice, but a solution exists.

**Comment by EngRajabi** [comment]
I checked. You can use this flag if the feature is not active but the warnings are active.
```csharp
<Nullable>warnings</Nullable>
```
